File: src/main/java/pt/haslab/dql/tictactoe/agents/Agent.java
package pt.haslab.dql.tictactoe.agents;

import pt.haslab.dql.tictactoe.game.Board;

/**
 * The abstract class of an Agent. An agent implementation
 * must provide a game strategy, i.e. to state
 * what move should be made by the agent depending on
 * the board state.
 * Created by nunomachado on 06/08/17.
 */
public abstract class Agent {
    public abstract String act(Board state);
}

File: tictactoe/agents/agent.py
from abc import ABC, abstractmethod

from tictactoe.game.board import Board


class Agent(ABC):
    """The abstract class of an Agent. An agent implementation
    must provide a game strategy, i.e. to state
    what move should be made by the agent depending on
    the board state.
    """

    @abstractmethod
    def make_move(self, state: Board) -> str:
        pass
